libraries/mbed/common/rtc_time.c

```diff
@@ -58,6 +75,15 @@ clock_t clock() {
     return t;
 }
 
+void attach_rtc(time_t (*read_rtc)(void), void (*write_rtc)(time_t), void (*init_rtc)(void), int (*isenabled_rtc)(void)) {
+    _rtc_read = read_rtc;
+    _rtc_write = write_rtc;
+    _rtc_init = init_rtc;
+    _rtc_isenabled = isenabled_rtc;
+}
```

Comment:
Init is called as it is even now. set_time reinits RTC, and time() invokes set_time() if RTC is not enabled. 
Invoke read/write, do mean HAL functions? Share the code snippet, might help.